packages/server/src/cli.ts

```diff
@@ -20,27 +20,41 @@ const collect = <T>(value: T, previous: T[]): T[] => {
 	return previous.concat(value);
 };
 
+const split = (value: string | undefined): string[] => {
+	if (!value) {
+		return [];
+	}
+	return value.split(',');
+}
+
+const envVarBool = (value: string | undefined, defaultVal: boolean): boolean => {
+	if (value) {
+		return value.toLowerCase() === 'true'
+	}
+	return defaultVal;
+}
+
 commander.version(process.env.VERSION || "development")
 	.name("code-server")
 	.description("Run VS Code on a remote server.")
 	.option("--cert <value>")
 	.option("--cert-key <value>")
-	.option("-e, --extensions-dir <dir>", "Override the main default path for user extensions.")
-	.option("--extra-extensions-dir [dir]", "Path to an extra user extension directory (repeatable).", collect, [])
-	.option("--extra-builtin-extensions-dir [dir]", "Path to an extra built-in extension directory (repeatable).", collect, [])
-	.option("-d, --user-data-dir <dir>", "Specifies the directory that user data is kept in, useful when running as root.")
-	.option("--data-dir <value>", "DEPRECATED: Use '--user-data-dir' instead. Customize where user-data is stored.")
-	.option("-h, --host <value>", "Customize the hostname.", "0.0.0.0")
-	.option("-o, --open", "Open in the browser on startup.", false)
+	.option("-e, --extensions-dir <dir>", "Override the main default path for user extensions.", process.env.EXTENSIONS_DIR)
+	.option("--extra-extensions-dir [dir]", "Path to an extra user extension directory (repeatable).", collect, split(process.env.EXTRA_EXTENSIONS_DIR))
+	.option("--extra-builtin-extensions-dir [dir]", "Path to an extra built-in extension directory (repeatable).", collect, split(process.env.EXTRA_BULTIN_EXTENSIONS_DIR))
+	.option("-d, --user-data-dir <dir>", "Specifies the directory that user data is kept in, useful when running as root.", process.env.USER_DATA_DIR)
+	.option("--data-dir <value>", "DEPRECATED: Use '--user-data-dir' instead. Customize where user-data is stored.", process.env.DATA_DIR)
```

Comment:
Deprecated flags shouldn't have env variables. We don't want people to use them anymore.